### PR TITLE
Hapi 19 support

### PR DESCRIPTION
**Pull request by terra-novalis**
Introduces fixes for Hapi 19 support and bumps major version to 4.0.0.

* Also removes "use strict" since ES6 functionality is in use

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 language: node_js
 
 node_js:
-  - 8
-  - 10
   - 12
```
Comment (fknop) [inline .travis.yml:4]: Node 14 just release, maybe we could add it as well

File: test/test.js [hunk 2]
```diff
@@ -218,14 +256,14 @@ const register = () => {
...
-          testDate: Joi.date(),
-          testArray: Joi.array(),
-          testObject: Joi.object(),
-          page: Joi.number(),
-          limit: Joi.number()
+          testDate: testJoi.date(),
+          testArray: testJoi.array(),
+          testObject: testJoi.object(),
+          page: testJoi.number(),
+          limit: testJoi.number()
```
Comment (fknop) [inline test/test.js:266]: This is just a change that's required in test, right?

File: .travis.yml (hunk 1 above)
Comment (terra-novalis) [reply .travis.yml:4]:
```suggestion
  - 12
  - 14
```

File: test/test.js (hunk 2 above)
Comment (terra-novalis) [reply test/test.js:266]: https://github.com/hapijs/joi/issues/2031

⬆️ Due to this change, by default, Joi will not automatically coerce strings containing arrays or objects into those data-types. The update to the test reflects that any package using `hapi-pagination` will have to accommodate this change (which would have been true anyway if they consumer package had already updated to Hapi >= 19).

> Joi.object() and Joi.array() no longer support string coercion. This means that a string value of '[1,2,3]' will no longer be automatically converted to a valid array when convert is true, and similarly for objects. This could be a problem if you support object or array strings in HTTP request query strings.

tl;dr: Yes, it's not necessary to use custom Joi extensions to use `hapi-pagination` as of this change, but it _will_ be necessary if you expect Joi to coerce data types for you.

**Review by fknop — APPROVED** [review]
